receiver/googlecloudspannerreceiver/internal/metadata/labelvalue.go

```diff
@@ -0,0 +1,215 @@
+// limitations under the License.
+
+package metadata
+
+import (
+	"sort"
+	"strings"
+)
+
+type LabelValueMetadata interface {
+	ValueMetadata
+}
+
+type LabelValue interface {
+	LabelValueMetadata
+	Value() interface{}
+}
+
+type queryLabelValueMetadata struct {
+	name       string
+	columnName string
+}
+
+func newQueryLabelValueMetadata(name string, columnName string) queryLabelValueMetadata {
+	return queryLabelValueMetadata{
+		name:       name,
+		columnName: columnName,
+	}
+}
+
+type StringLabelValueMetadata struct {
+	queryLabelValueMetadata
+}
+
+func NewStringLabelValueMetadata(name string, columnName string) StringLabelValueMetadata {
+	return StringLabelValueMetadata{
+		queryLabelValueMetadata: newQueryLabelValueMetadata(name, columnName),
+	}
+}
+
+type Int64LabelValueMetadata struct {
+	queryLabelValueMetadata
+}
+
+func NewInt64LabelValueMetadata(name string, columnName string) Int64LabelValueMetadata {
+	return Int64LabelValueMetadata{
+		queryLabelValueMetadata: newQueryLabelValueMetadata(name, columnName),
+	}
+}
+
+type BoolLabelValueMetadata struct {
+	queryLabelValueMetadata
+}
+
+func NewBoolLabelValueMetadata(name string, columnName string) BoolLabelValueMetadata {
+	return BoolLabelValueMetadata{
+		queryLabelValueMetadata: newQueryLabelValueMetadata(name, columnName),
+	}
+}
+
+type StringSliceLabelValueMetadata struct {
+	queryLabelValueMetadata
+}
+
+func NewStringSliceLabelValueMetadata(name string, columnName string) StringSliceLabelValueMetadata {
+	return StringSliceLabelValueMetadata{
+		queryLabelValueMetadata: newQueryLabelValueMetadata(name, columnName),
+	}
+}
+
+type ByteSliceLabelValueMetadata struct {
+	queryLabelValueMetadata
+}
+
+func NewByteSliceLabelValueMetadata(name string, columnName string) ByteSliceLabelValueMetadata {
+	return ByteSliceLabelValueMetadata{
+		queryLabelValueMetadata: newQueryLabelValueMetadata(name, columnName),
+	}
+}
+
+type stringLabelValue struct {
+	StringLabelValueMetadata
+	value string
+}
+
+type int64LabelValue struct {
+	Int64LabelValueMetadata
+	value int64
+}
+
+type boolLabelValue struct {
+	BoolLabelValueMetadata
+	value bool
+}
+
+type stringSliceLabelValue struct {
+	StringSliceLabelValueMetadata
+	value string
+}
+
+type byteSliceLabelValue struct {
+	ByteSliceLabelValueMetadata
+	value string
+}
+
+func (metadata queryLabelValueMetadata) Name() string {
+	return metadata.name
+}
+
+func (metadata queryLabelValueMetadata) ColumnName() string {
+	return metadata.columnName
+}
+
+func (metadata StringLabelValueMetadata) ValueHolder() interface{} {
+	var valueHolder string
+
+	return &valueHolder
+}
+
+func (value stringLabelValue) Value() interface{} {
+	return value.value
+}
+
+func newStringLabelValue(metadata StringLabelValueMetadata, valueHolder interface{}) stringLabelValue {
+	return stringLabelValue{
+		StringLabelValueMetadata: metadata,
+		value:                    *valueHolder.(*string),
```

Comment:
Does it make sense to get this into an assignment statement and check for errors? Same question applies to the other similar places.

Reply:
We do not expect that this will be used somewhere else, except for internal usage and metadata parsing from metadata configuration, you'll see this in upcoming requests. That is why I do not perform additional checks.